server.py

```diff
@@ -559,6 +561,28 @@ async def post_history(request):
                     self.prompt_queue.delete_history_item(id_to_delete)
 
             return web.Response(status=200)
+        
+        @routes.post("/download")
+        async def download_handler(request):
+            async def report_progress(filename: str, status: DownloadStatus):
+                await self.send_json(filename, {
```

Comment:
I don't think we should send filename as type here as it can collide with existing message types.

Reply:
I updated it to use the key `download_progress`, and the filename will be a field in the event.